backend/tarsy/services/events/event_helpers.py

```
@@ -286,3 +289,47 @@ async def publish_stage_completed(
+    Args:
+        session_id: Session identifier
+    """
+    try:
+        async_session_factory = get_async_session_factory()
+        async with async_session_factory() as session:
+            event = SessionCancelledEvent(session_id=session_id, status=AlertSessionStatus.CANCELLED.value)
+            # Publish to global 'sessions' channel for dashboard
+            await publish_event(session, EventChannel.SESSIONS, event)
+            # Also publish to session-specific channel for detail views
+            await publish_event(session, f"session:{session_id}", event)
+            logger.info(f"[EVENT] Published session.cancelled to channels")
```

Ruff (0.14.1)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)